python/Ganga/Core/GangaRepository/Registry.py

```diff
@@ -788,7 +768,8 @@ def _flush(self, _objs=None):
             for obj_id in obj_ids:
                 self.unlock_transaction(obj_id)
 
-            self._lock.release()
+            for obj in objs:
+                obj._releaseParentLock()
```

Comment:
No, I want to prevent stupid oddities like the registry trying to load the object and flush it from 2 threads. This shouldn't happen but the other locks will only mean the resulting mess  is consistent at the atomic level of single get/set actions. the XML repo (specifically VStreamer functions) subvert some of the setter methods by design.